src/libraries/System.Text.Json/src/System/Collections/Generic/StackExtensions.netstandard.cs

```diff
@@ -3,34 +3,35 @@
 // See the LICENSE file in the project root for more information.
 
 using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
 using System.Runtime.CompilerServices;
 
 namespace System.Collections.Generic
 {
     /// <summary>Polyfills for <see cref="Stack{T}"/>.</summary>
     internal static class StackExtensions
```

Comment:
We annotated this even though its only included for netstandard. Is that expected?

What about the codes paths that are if/def'd out when !BUILDING_INBOX_LIBRARY?
https://github.com/dotnet/runtime/blob/4f9ae42d861fcb4be2fcd5d3d55d5f227d30e723/src/libraries/System.Text.Json/src/System.Text.Json.csproj#L8

For example:
https://github.com/dotnet/runtime/blob/57e57556ebe7e58f2fd54759725246bd48045301/src/libraries/System.Text.Json/src/System/Text/Json/Node/JsonObject.cs#L156-L173

Reply:
Had nullability related error on CI netstandard build `A default expression introduces a null value when 'T' is a non-nullable reference type.` Anyways i would assume all code paths should be annotated accordingly so that all build types/versions were successful. In some netstandard versions and full framework Debug.Assert(value != null); doesn't suppress the build error where we need to bang even we had asserted

Reply:
Since we already have a `NoWarn` for nullability for non-netcoreapp TFMs, should we disable the warnings for this particular file rather than annotating it?

**I am also fine with leaving things as is**, so this is primarily just a question to understand the different options.

Reply:
Well `[MaybeNullWhen(false)]` on the method signature could be useful for calling code to be aware of null ref. But i see all current usages calling it with struct type so could say not useful for now but not sure if it will be used with ref type ever

Reply:
> but not sure if it will be used with ref type ever

I wouldn't assume it will be used with just struct types. Using `default!;` is fine.